src/components/molecules/DataWrapper.tsx

```diff
@@ -0,0 +1,22 @@
+import React, { ReactNode, useEffect, useState } from 'react';
+import { ApolloError, DocumentNode, useQuery } from '@apollo/client';
+import { TimeEntry } from '../atoms/ThresholdChart';
+import { formatMetData } from '../../utils/formatMetData';
+
+export type DataWrapperProps = {
+    query: DocumentNode;
+    children: (args: { data: TimeEntry[] | undefined; loading: boolean; error: ApolloError | undefined }) => ReactNode;
+};
+
+const DataWrapper: React.FC<DataWrapperProps> = ({ children, query }) => {
+    const [formattedData, setFormattedData] = useState<TimeEntry[] | undefined>(undefined);
+    const { loading, error, data } = useQuery(query);
+
+    useEffect(() => {
+        if (data) setFormattedData(formatMetData(data));
+    }, [data]);
+
+    return <>{children({ data: formattedData, loading, error })}</>;
```

Comment:
Since `data` will only change once, you could structure it like this (presuming you handle `loading`/`error` in this component as well)

```tsx
if (loading) {
  return <Loading />;
}

if (error) {
  return <Error error={error} />;
}

return children(data);
```

This would make typing the child a lot easier.

Reply:
How do you return just `children(data)`. I can’t get it to work without wrapping it in `<>{children(data)}</>`

Reply:
I don't know, I just made something up.

You can try
- looking at the stack trace. what went wrong?
- what type did it produce, and what does it expect?
- googling the issue
- taking a look at the typing in other examples

Reply:
For reference, you're wrapping the child function invocation in a fragment. To understand why it doesn't work without, it might be helpful to understand what this does. 

It might also be helpful to research these kinds of issues before asking in the code review. If you're still stuck after some time, you can in #help-programming in Slack. I'd appreciate it if we could (for the most part) try to keep code reviews to discussions rather than programming questions.